File: react-native-counter/ios/src/Counter.m
#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(Counter, NSObject)

RCT_EXTERN_METHOD(initialize:(RCTPromiseResolveBlock)resolve reject:(RCTPromiseRejectBlock)reject );
RCT_EXTERN_METHOD(test:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject );
RCT_EXTERN_METHOD(testProperty);

@end

File: react-native-counter/ios/src/Counter.swift
//
//  Counter.swift
//  react-native-counter
//
//  Created by Yunfeng Lin on 12/5/19.
//

import Foundation
import AWSDK

@objc(Counter)
class Counter: NSObject {
  @objc var testProperty: String = "hello"
  
  @objc func requiresMainQueueSetup () -> Bool {
    return false
  }

  @objc(test:rejecter:) func test (_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
    resolve(true)
  }
  
  @objc func constToExport () -> [String: Any] {
    return ["name": "Counter"]
  }
    
    @objc func initialize (_ resolve: @escaping RCTPromiseResolveBlock, reject: @escaping RCTPromiseRejectBlock) {
      AWSDKService.initialize(withLaunchParams: [kAWSDKBundleID: "com.uhg.amwelldemo",
                                                 kAWSDKUrl: "https://iot58.amwellintegration.com/",
                                                 kAWSDKKey: "bf8a5665"]) { (success, error) in
                                                  if success {
                                                    resolve("AWSDKService initialize successfully!");
                                                  } else {
                                                    reject("init_fail", "looks like init failed!", error);
                                                  }
                                                  
      }
    }
}
